_posts/2017-03-30-storage-formats-quickstep.markdown

```diff
@@ -0,0 +1,140 @@
+
+Putting it all together, the picture below shows the compressed column store format in which the compression is applied on the ID column. 
+
+![Column Store with Compression](../assets/storage-format-compressed-column-store.jpg)
+
+Observe that the memory occupied by the three tuples together is only 43 bytes, compared to 48 bytes in the row store and the column store formats. As the number of tuples increase, there may be more opportunities for compression and thus more memory savings.
+
+There are some aspects related to storage management which deserve a blog post of their own. This includes the block-based storage design, and the impact of the above storage formats on various analytical queries. I hope to write more blog posts to cover these topics in the future.
+
+<h1>Creating tables with various storage formats in Quickstep</h1>
+
+Now I will show you how to use the above storage formats to create tables in Quickstep. The storage format specification is part of the `CREATE TABLE SQL` command. We will continue with our toy example above to illustrate the various ways in which we can use the storage formats in Quickstep. 
+
+{% highlight sql %}
+CREATE TABLE employee (
+ID INT NOT NULL, 
+Age INT NOT NULL, 
+Name VARCHAR(8) NOT NULL
+) WITH BLOCKPROPERTIES (
+  TYPE rowstore,
+  BLOCKSIZEMB 4);
+{% endhighlight %}
+
+I will now describe some keywords used in the above SQL statement. The keyword `BLOCKPROPERTIES` refers to the storage properties of this table. The above command means that all the blocks in the employee table will use the row store format, with each block sized to a maximum of 4 MB.
+
+{% highlight sql %}
+CREATE TABLE employee (
+ID INT NOT NULL, 
+Age INT NOT NULL, 
+Name VARCHAR(8) NOT NULL
+) WITH BLOCKPROPERTIES (
+  TYPE compressed_columnstore,
+  SORT ID,
+  COMPRESS ID,
+  BLOCKSIZEMB 4);
+{% endhighlight %}
+
+In this case, all the blocks in the employee table have compressed column store as their storage format. The values in each block are sorted by the `ID` values, and the compression is applied on the `ID` column (just like we described earlier).
+
+The `COMPRESS` keyword accepts one or more or `ALL` the columns from the table. Thus it can look like:
+
+{% highlight sql %}
+CREATE TABLE employee (
+ID INT NOT NULL, 
+Age INT NOT NULL, 
+Name VARCHAR(8) NOT NULL
+) WITH BLOCKPROPERTIES (
+  TYPE compressed_columnstore,
+  SORT ID,
+  COMPRESS (ID, Age),
+  BLOCKSIZEMB 4);
+{% endhighlight %}
+
+The above command indicates that compression will be applied on the `ID` and the `Age` columns only.
+
+Finally, the command below indicates that the compression is applied on all the columns.
+
+{% highlight sql %}
+CREATE TABLE employee (
+ID INT NOT NULL, 
+Age INT NOT NULL, 
+Name VARCHAR(8) NOT NULL
+) WITH BLOCKPROPERTIES (
+  TYPE compressed_columnstore,
+  SORT ID,
+  COMPRESS ALL,
```

Comment:
Oh I see that you use ALL here. Maybe in the previous command specify: `COMPRESS (Age, Name)`